spring-2013/hw1/src/CreditCardValidator.hs

```diff
@@ -0,0 +1,22 @@
+module CreditCardValidator
+    ( toDigits
+    , doubleEveryOther
+    , sumDigits
+    , validate
+    ) where
+
+toDigits :: Integer -> [Integer]
+toDigits n
+    | n <= 0 = []
+    | otherwise = map readChar . show $ n
+    where readChar = read . (:[])
+
+doubleEveryOther :: [Integer] -> [Integer]
+doubleEveryOther = reverse . zipWith (*) (cycle [1, 2]) . reverse
+
+sumDigits :: [Integer] -> Integer
+sumDigits = sum . (toDigits =<<)
+
+validate :: Integer -> Bool
+validate = checkRemainder . sumDigits . doubleEveryOther . toDigits
+    where checkRemainder = (== 0) . (`mod` 10)
```

Comment:
To be honest, I kinda like this named as an action. Reading this as `check remainder after summing the digits after doubling every other after converting to digits` feels right